src/util/modbusdatatype.h

```diff
@@ -0,0 +1,161 @@
+#ifndef MODBUSDATATYPE_H
+#define MODBUSDATATYPE_H
+
+#include <QObject>
+
+class ModbusDataType
+{
+
+public:
+
+    enum class Type
+    {
+        UNSIGNED_16 = 0,
```

Comment:
Not a problem, but just as an optional improvement. You could introduce an `UNKNOWN` type as value `0`. Then you can use it directly as such: `if (!type)`. 

Additionally this could remove the need for the `bOk` boolean primitive you are using in some places.

Reply:
I agree that it would provide some benefits, like removing the bOk argument and simplifying things for the future.

But actually I like the explicitness of the extra argument at the moment. This reminds me that I should check the result. If the enum is expanded with a UNKNOWN then I also need to handle this value everywhere where i use this type.

And, this is the way it is also done in Qt function: https://doc.qt.io/qt-6/qstring.html#toDouble

Reply:
No problem, I'll raise a review comment at Qt as well then 😜